usaspending_api/agencies/models/raw_agency_codes_csv.py

```diff
@@ -0,0 +1,107 @@
+                "admin_org": prep(agency["ADMIN_ORG"]),
+                "toptier_flag": strtobool(prep(agency["TOPTIER_FLAG"])),
+                "is_frec": strtobool(prep(agency["IS_FREC"])),
+                "frec_cgac_association": strtobool(prep(agency["FREC CGAC ASSOCIATION"])),
+                "user_selectable_on_usaspending_gov": strtobool(prep(agency["USER SELECTABLE ON USASPENDING.GOV"])),
+                "mission": prep(agency["MISSION"]),
+                "website": prep(agency["WEBSITE"]),
+                "congressional_justification": prep(agency["CONGRESSIONAL JUSTIFICATION"]),
+                "icon_filename": prep(agency["ICON FILENAME"]),
+                "comment": prep(agency["COMMENT"]),
+            }
+            for row_number, agency in enumerate(agencies)
+        ]
+
+        if force is not True:
+            existing_agencies = list(RawAgencyCodesCSV.objects.all().order_by("row_number").values())
+            if agencies == existing_agencies:
+                return 0
+
+        with transaction.atomic():
+            self.get_queryset().all().delete()
+            self.bulk_create([RawAgencyCodesCSV(**agency) for agency in agencies])
+
+        return len(agencies)
+
+
+class RawAgencyCodesCSV(Model):
+    """
+    A copy of the raw agency_codes.csv file as maintained by the product owner.
+
+    I would recommend never joining directly to this table from production code and never pointing
+    hard foreign keys at it.  Always carve off what you need into independent tables.  I would even
+    steer away from views of any sort based on this table as raw tables are more likely to change
+    and views prevent structural changes to their underlying data sources.  Independent tables are
+    your most flexible option.
+    """
+    row_number = IntegerField(primary_key=True)
+    cgac_agency_code = TextField(blank=True, null=True)
+    agency_name = TextField(blank=True, null=True)
+    agency_abbreviation = TextField(blank=True, null=True)
+    frec = TextField(blank=True, null=True)
+    frec_entity_description = TextField(blank=True, null=True)
+    frec_abbreviation = TextField(blank=True, null=True)
+    subtier_code = TextField(blank=True, null=True)
+    subtier_name = TextField(blank=True, null=True)
+    subtier_abbreviation = TextField(blank=True, null=True)
+    admin_org_name = TextField(blank=True, null=True)
+    admin_org = TextField(blank=True, null=True)
+    toptier_flag = BooleanField()
+    is_frec = BooleanField()
+    frec_cgac_association = BooleanField()
+    user_selectable_on_usaspending_gov = BooleanField()
+    mission = TextField(blank=True, null=True)
+    website = TextField(blank=True, null=True)
+    congressional_justification = TextField(blank=True, null=True)
+    icon_filename = TextField(blank=True, null=True)
+    comment = TextField(blank=True, null=True)
+
+    objects = RawAgencyCodesCSVManager()
+
+    class Meta:
+        db_table = "raw_agency_codes_csv"
```

Comment:
It would be nice to agree to a naming convention for "raw"-ish data.

On a side note, I keep going back and forth on whether or not to even keep this file in the database.  On the one hand, it would be nice to have it for easy reference.   On the other, I'm not sure I want people glomming onto a raw table which is a risk whenever something is included in the database.

I tried putting it in its own schema, but Django really doesn't want to easily support that functionality.  I feel like this would give a cleaner separation and make it more obvious that it's a different kind of animal.

Anyhow, back to the original question, I'd like to differentiate between "raw" data and "reference" data.  "raw"-ish data is what this table is.  It contains warts and would not be something I'd want to query regularly.  Normally this would go into a "staging" schema or some place where we can work on it in isolation without risk of it become part of the production database.  This "raw" table is carved into a bunch of "reference" tables.

Anyhow, I'll rename it `raw_agency`... or maybe I'll remove it altogether.  AARRGGHH!!!

Reply:
Removal is better. If we don't want anybody accessing this table